.github/workflows/deploy.yml

```diff
@@ -0,0 +1,135 @@
+name: Deploy
+
+on:
+  push:
+    branches: [ '**' ]
+
+jobs:
+
+
+ versionning:
+    name: Versionning
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.genver.outputs.version }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ github.ref }}
+        fetch-depth: 0
+
+    - name: Generate Version
+      id: genver
+      run: |
+        VERSION=$(docker run --rm -v $(pwd):/repo codacy/git-version /bin/git-version --folder=/repo --release-branch=${{ github.ref_name }}-pre --dev-branch=main)
+        echo "version=$VERSION" >> $GITHUB_OUTPUT
+        echo "VERSION : $VERSION"
+        echo "Version :" >> $GITHUB_STEP_SUMMARY
+        echo "$VERSION" >> $GITHUB_STEP_SUMMARY
+
+ format:
```

Comment:
I'd put formatting job in its own worflow file with a trigger on pull requests as it is done in ArmoniK.Core

Reply:
Done !